dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3107,6 +3107,44 @@ void Sedp::data_received(DCPS::MessageId message_id,
 }
 #endif
 
+//This is a function to unify the notifcation of liveliness within RTPS
+//The local participant map is checked for associated entities and then they are notified
+//of liveliness if their QoS is compatibile
+void
+Sedp::notify_liveliness(const ParticipantMessageData& pmd)
+{
+  const RepoId& guid = pmd.participantGuid;
+  RepoId guid_participant = guid;
+  guid_participant.entityId = ENTITYID_PARTICIPANT;
+  RepoId prefix = pmd.participantGuid;
+  prefix.entityId = EntityId_t(); // Clear the entityId so lower bound will work.
+  bool is_automatic = true;
+  ACE_DEBUG((LM_DEBUG, "(%P|%t) %T Sedp::notify_liveliness(const ParticipantMessageData& pmd), %d\n",
```

Comment:
Make conditional on log level or flag or something.
Do you need the function signature?
Instead of logging the identity, log the entire guid with `LogGuid(guid).c_str()`.